src/irmin-pack/unix/file_manager.ml

```diff
@@ -749,4 +760,44 @@ struct
     | T15 ->
         (* Unreachable *)
         assert false
+
+  (* TODO : this does not work yet with GC. *)
+  let split t : (unit, [> Errs.t ]) result =
+    let open Result_syntax in
+    (* Step 1. Fsync last chunk *)
+    let* () = Suffix.fsync t.suffix in
+
+    (* Step 2. Create a new chunk file *)
+    let auto_flush_threshold =
+      match Suffix.auto_flush_threshold t.suffix with
+      | None -> assert false
+      | Some x -> x
+    in
+    let cb _ = suffix_requires_a_flush_exn t in
+    let* () =
+      Suffix.add_chunk ~auto_flush_threshold
+        ~auto_flush_procedure:(`External cb) t.suffix
+    in
+
+    (* Step 3. Update the control file *)
+    let* () =
+      let pl = Control.payload t.control in
+      let pl =
+        let open Payload in
+        let chunk_num = pl.chunk_num + 1 in
```

Comment:
In my GC prototyping, I found it useful to expose `Chunk_suffix.chunk_num`. I think it would be nice to use that here. Either you can add this to your PR (and increment it in `add_chunk`) or we can clean it up later.

Reply:
since `Chunk_suffix.chunk_num` is not exposed yet, I left the code as is, if that's ok.